packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -117,9 +117,24 @@ async function handleRequest(
 	const site = config.buildOptions.site ? new URL(config.buildOptions.site) : undefined;
 	const devRoot = site ? site.pathname : '/';
 	const origin = `${viteServer.config.server.https ? 'https' : 'http'}://${req.headers.host}`;
+	const buildingToSSR = !!config._ctx.adapter?.serverEntrypoint;
 	const url = new URL(origin + req.url);
 	const pathname = decodeURI(url.pathname);
 	const rootRelativeUrl = pathname.substring(devRoot.length - 1);
+	if(!buildingToSSR) {
+		// Prevent user from depending on search params when not doing SSR.
+		for(const [key] of url.searchParams) {
+			url.searchParams.delete(key);
+		}
+	}
+
+	// Headers are only available when using SSR.
+	const request = createRequest({
```

Comment:
TIL that Node.js `http.IncomingMessage` wasn't a `Request` object? For some reason I thought that that had always been the case.